### PR TITLE
Azure Firewall integration

### PR DESCRIPTION
**Pull request by ebeahan**
## What does this PR do?

This PR adds a new integration for Azure Firewall.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

```
cd packages/azure
elastic-package test pipeline
```

## Related issues

Relates to #1205

## Screenshots

Screen Shot 2022-06-23 at 3 13 20 PM

Screen Shot 2022-06-23 at 3 15 21 PM

filebeat-azure-firewall-overview

filebeat-azure-firewall-dns-proxy

### CODE REVIEW

File: packages/azure/data_stream/firewall_logs/elasticsearch/ingest_pipeline/azure-shared-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+description: Pipeline for parsing Azure shared log fields.
+processors:
+  - set:
+      field: cloud.provider
+      value: azure
+  - grok:
+      field: azure.resource_id
+      patterns:
+      - /SUBSCRIPTIONS/%{SUBID:azure.subscription_id}/RESOURCEGROUPS/%{GROUPID:azure.resource.group}/PROVIDERS/%{PROVIDERNAME:azure.resource.provider}/NAMESPACES/%{NAMESPACE:azure.resource.namespace}/AUTHORIZATIONRULES/%{RULE:azure.resource.authorization_rule}
+      - /subscriptions/%{SUBID:azure.subscription_id}/resourceGroups/%{GROUPID:azure.resource.group}/providers/%{PROVIDERNAME:azure.resource.provider}/namespaces/%{NAMESPACE:azure.resource.namespace}/authorizationRules/%{RULE:azure.resource.authorization_rule}
```
Comment (efd6) [inline packages/azure/data_stream/firewall_logs/elasticsearch/ingest_pipeline/azure-shared-pipeline.yml:11]:
```suggestion
      - (?i)/subscriptions/%{SUBID:azure.subscription_id}/resourceGroups/%{GROUPID:azure.resource.group}/providers/%{PROVIDERNAME:azure.resource.provider}/namespaces/%{NAMESPACE:azure.resource.namespace}/authorizationRules/%{RULE:azure.resource.authorization_rule}
```

Could simplify the char classes below similarly. Also in the grok processors below.

File: packages/azure/data_stream/firewall_logs/sample_event.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
+{
```
Comment (efd6) [inline packages/azure/data_stream/firewall_logs/sample_event.json:1]: This also looks like it's not from a recent version. `elastic-package test system -g`?

File: packages/azure/data_stream/firewall_logs/_dev/test/pipeline/test-applicationrules-raw.log-expected.json [hunk 3]
```diff
@@ -0,0 +1,492 @@
+{
```
[the remark this answers is not in the document]
Comment (ebeahan) [reply packages/azure/data_stream/firewall_logs/_dev/test/pipeline/test-applicationrules-raw.log-expected.json:1]: Addressed.

File: packages/azure/data_stream/firewall_logs/agent/stream/azure-eventhub.yml.hbs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+storage_account_key: {{storage_account_key}}
+{{/if}}
+{{#if resource_manager_endpoint}}
+resource_manager_endpoint: {{resource_manager_endpoint}}
+{{/if}}
+tags:
+{{#if preserve_original_event}}
+  - preserve_original_event
+{{/if}}
+{{#each tags as |tag i|}}
```
[the remark this answers is not in the document]
Comment (ebeahan) [reply packages/azure/data_stream/firewall_logs/agent/stream/azure-eventhub.yml.hbs]: Good catch! I've removed these.

File: packages/azure/data_stream/firewall_logs/elasticsearch/ingest_pipeline/azure-shared-pipeline.yml (hunk 1 above)
Comment (ebeahan) [reply packages/azure/data_stream/firewall_logs/elasticsearch/ingest_pipeline/azure-shared-pipeline.yml]: Good suggestion - simplified in https://github.com/elastic/integrations/pull/3578/commits/8e3aecf79ddacd354a8af66aca85274512d8e9dd

File: packages/azure/data_stream/firewall_logs/fields/fields.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    - name: icmp
+      type: group
+      fields:
+        - name: request
+          type: group
+          fields:
+            - name: code
+              type: keyword
+              description: |
+                ICMP request code
```
[the remark this answers is not in the document]
Comment (ebeahan) [reply packages/azure/data_stream/firewall_logs/fields/fields.yml:44]: Unless a numeric field needs to support `range` queries, ES [recommends](https://www.elastic.co/guide/en/elasticsearch/reference/current/keyword.html) `keyword` for the better `term` query performance.

For ICMP messages, I think users are more likely query for a specific request code than across a range.

**Comment by ebeahan** [comment]
Last round of comments have been addressed - ready for another review round.

File: packages/azure/_dev/build/docs/firewall_logs.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+- The logs have to be [exported first to the event hub](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-create-kafka-enabled).
+- To export activity logs to event hubs users can follow the steps [here](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/activity-log-export).
+- To export audit and sign-in logs to event hubs users can follow the steps [here](https://docs.microsoft.com/en-us/azure/active-directory/reports-monitoring/tutorial-azure-monitor-stream-logs-to-event-hub).
+
+## Settings
+
+`eventhub` :
+  _string_
+It is a fully managed, real-time data ingestion service. Elastic recommends using only letters, numbers, and the hyphen (-) character for Event Hub names to maximize compatibility. You can use existing Event Hubs having underscores (_) in the Event Hub name; in this case, the integration will replace underscores with hyphens (-) when it uses the Event Hub name to create dependent Azure resources behind the scenes (e.g., the storage account container to store Event Hub consumer offsets).
```
Comment (efd6) [inline packages/azure/_dev/build/docs/firewall_logs.md:15]: s/It/An Event Hub/?

File: packages/azure/_dev/build/docs/firewall_logs.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+Default value `insights-operational-logs`.
+
+`consumer_group` :
+_string_
+ The publish/subscribe mechanism of Event Hubs is enabled through consumer groups. A consumer group is a view (state, position, or offset) of an entire event hub. Consumer groups enable multiple consuming applications to each have a separate view of the event stream, and to read the stream independently at their own pace and with their own offsets.
+Default value: `$Default`
+
+`connection_string` :
+_string_
+The connection string required to communicate with Event Hubs, steps [here](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-get-connection-string).
```
Comment (efd6) [inline packages/azure/_dev/build/docs/firewall_logs.md:25]: s/string required/string is required/
s/steps/see steps/

File: packages/azure/_dev/build/docs/firewall_logs.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+A Blob Storage account is required in order to store/retrieve/update the offset or state of the eventhub messages. This means that after stopping the Azure logs package it can start back up at the spot that it stopped processing messages.
+
+`storage_account` :
+_string_
+The name of the storage account the state/offsets will be stored and updated.
```
Comment (efd6) [inline packages/azure/_dev/build/docs/firewall_logs.md:31]: s/account the/account where the/

**Review by efd6 — APPROVED** [review]
LGTM

**Comment by ebeahan** [comment]
@elastic/obs-cloud-monitoring would you also please review as package owners?

**Pull request merged**